log.cpp

```diff
@@ -0,0 +1,79 @@
+#include "log.hpp"
+#include <iostream>
+#include <fstream>
+#include <ctime>
+using namespace std;
+
+Log::Log(std::string path) {
+	this->path = path;
+	logMsg.open(path);

```

Comment:
You need to check if the file exists and whether it has actually opened. Have a look at FileSystem.cpp to see how i did it